### PR TITLE
Goal view polish

### PR DESCRIPTION
**Pull request by amitbharadwaj**
Polished Goal View

### CODE REVIEW

File: res/layout/activity_goal_details.xml [hunk 1]
```diff
@@ -6,142 +6,199 @@
...
-            style="@style/MediumAmount"
-            android:layout_alignRight="@+id/tvDueDateLabel"
-            android:layout_alignTop="@+id/tvPaymentDue"
-            android:text="@string/sample_due_date"
-            android:textAppearance="?android:attr/textAppearanceMedium" />
+            android:layout_above="@+id/tvPaymentDue"
+            android:layout_alignRight="@+id/tvPaymentDue"
+            android:text="@string/next_payment_label" />
+
+        <!-- <TextView -->
```
Comment (josemontesdeoca) [inline res/layout/activity_goal_details.xml]: If this is no longer needed, please remove.

File: res/layout/activity_goal_details.xml [hunk 2]
```diff
@@ -6,142 +6,199 @@
...
+            android:progressDrawable="@drawable/goal_payment_progress_bar" />
+    </RelativeLayout>
 
-        <ListView
-            android:id="@+id/lvPastPayments"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content" >
-        </ListView>
-    </LinearLayout>
+<!--     <LinearLayout -->
```
Comment (josemontesdeoca) [inline res/layout/activity_goal_details.xml]: If layout in comment is no longer needed, please remove

File: src/org/missionassetfund/apps/android/activities/AddGoalPaymentActivity.java [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+        transaction.setGoal(goal);
+        transaction.setTransactionDate(new Date());
+        transaction.setType(TransactionType.CREDIT);
+        transaction.saveInBackground(new SaveCallback() {
+
+            @Override
+            public void done(ParseException e) {
+                if (e == null) {
+                    Toast.makeText(AddGoalPaymentActivity.this, "Done savign txn",
```
Comment (josemontesdeoca) [inline src/org/missionassetfund/apps/android/activities/AddGoalPaymentActivity.java]: Text should live under [strings.xml](https://github.com/CodePath-MAF/AndroidClient/blob/master/res/values/strings.xml#L56)

File: src/org/missionassetfund/apps/android/activities/AddGoalPaymentActivity.java [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+        actionBar.setDisplayHomeAsUpEnabled(true);
+
+        lvSteps = (ListView) findViewById(R.id.lvSteps);
+
+        inputs = new ArrayList<Input>();
+        aInput = new InputItemAdapter(this, inputs);
+
+        // setup input steps
+        inputElements = new Input[] {
+                new Input("Amount Due", "0", 0, AmountInputFragment.class),
```
Comment (josemontesdeoca) [inline src/org/missionassetfund/apps/android/activities/AddGoalPaymentActivity.java:71]: Amount due needs to display what the user 'should' pay based on their payment amount

File: src/org/missionassetfund/apps/android/activities/AddGoalPaymentActivity.java [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+    private void showFragment(Class activeFragmentClass) {
+        FragmentManager mgr = getSupportFragmentManager();
+        FragmentTransaction transaction = mgr.beginTransaction();
+        try {
+            for (Input input : inputElements) {
+                Class klass = input.getFragmentClass();
+                Fragment fragment = mgr.findFragmentByTag(klass.getName());
+                if (klass == activeFragmentClass) {
+                    // Change ActionBar title
+                    setActionBarTitle(input);
```
Comment (josemontesdeoca) [inline src/org/missionassetfund/apps/android/activities/AddGoalPaymentActivity.java:117]: For you particular case the title should be "Amount". this is picking the 'name' property from input which is valid use case on the other 2 entry points coded. but here it does not change.

File: src/org/missionassetfund/apps/android/activities/AddGoalPaymentActivity.java [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+    private int getInputPosition(Class<? extends Fragment> fragmentClass) {
+        for (Input i : inputElements) {
+            if (i.getFragmentClass() == fragmentClass) {
+                return i.getPos();
+            }
+        }
+        return 0;
+    }
+
+    // private Class<?> getNextFragmentClass(Input input) {
```
Comment (josemontesdeoca) [inline src/org/missionassetfund/apps/android/activities/AddGoalPaymentActivity.java:235]: If the code is no longer used please remove

File: src/org/missionassetfund/apps/android/activities/GoalDetailsActivity.java [hunk 7]
```diff
@@ -26,36 +31,64 @@
...
-    TextView tvDueDate;
+    // TextView tvDueDate;
     TextView tvDueDateHuman;
-    TextView tvTotalTargetPayment;
-    TextView tvTargetDate;
-    TextView tvTargetDateHuman;
+    TextView tvNumPayments;
+    ProgressBar pbGoalPayment;
+
+    // TextView tvTotalTargetPayment;
```
Comment (josemontesdeoca) [inline src/org/missionassetfund/apps/android/activities/GoalDetailsActivity.java]: Please remove any code that is commented if no longer needed.

File: res/layout/activity_goal_details.xml [hunk 8]
```diff
@@ -6,142 +6,199 @@
...
-
-        <TextView
-            android:id="@+id/tvDueDate"
-            style="@style/MediumAmount"
-            android:layout_alignRight="@+id/tvDueDateLabel"
-            android:layout_alignTop="@+id/tvPaymentDue"
-            android:text="@string/sample_due_date"
-            android:textAppearance="?android:attr/textAppearanceMedium" />
+            android:layout_above="@+id/tvPaymentDue"
+            android:layout_alignRight="@+id/tvPaymentDue"
```
Comment (josemontesdeoca) [inline res/layout/activity_goal_details.xml]: this should not be alignRight. this is setting the right side fo the textview to align with the right side of tvPaymentDue and it does not show in the middle.

Advice: set up a vertical linear layout with gravity center and all items inside the linear layout will be centered automatically.

File: src/org/missionassetfund/apps/android/activities/GoalDetailsActivity.java [hunk 9]
```diff
@@ -92,32 +125,58 @@ protected void setupPaymentsAdapter() {
...
-
     }
 
     private void populateViews() {
         // Once goal is available let's setup views
-        tvTotalTargetPayment.setText(Double.toString(goal.getAmount()));
-        tvTargetDate.setText(FormatterUtils.formatMonthDate(goal.getGoalDate()));
-        tvTargetDateHuman.setText(FormatterUtils.getRelativeTimeHuman(goal.getGoalDate()));
+        // If goal is of type lending circle then change visibility to visible
+        // rlLendingCircle.setVisibility(View.GONE);
```
Comment (josemontesdeoca) [inline src/org/missionassetfund/apps/android/activities/GoalDetailsActivity.java]: I've asked on slack to see how we should check for this. in the mean time we can have a enum type property which Lending circle is 1 and we check for this type to show the faces.

**Comment by josemontesdeoca** [comment]
FYI: we were missing few bits of data on the view. https://files.slack.com/files-pri/T02BPJ5C9-F02DWSPAB/example.png

**Comment by fcduarte** [comment]
@amitbharadwaj why did you have to add the whole library? Adding just the jar didn't do the job?

File: res/layout/activity_goal_details.xml (hunk 1 above)
Comment (amitbharadwaj) [inline res/layout/activity_goal_details.xml]: Will do.

File: res/layout/activity_goal_details.xml (hunk 2 above)
Comment (amitbharadwaj) [inline res/layout/activity_goal_details.xml]: Will do

File: src/org/missionassetfund/apps/android/activities/AddGoalPaymentActivity.java (hunk 4 above)
Comment (amitbharadwaj) [inline src/org/missionassetfund/apps/android/activities/AddGoalPaymentActivity.java:71]: I have changed it below in setupDataFromIntent method. However, I do need to change the default amount in add payment fragment with payment amount. I will do this change.

File: res/layout/activity_goal_details.xml (hunk 8 above)
Comment (amitbharadwaj) [inline res/layout/activity_goal_details.xml]: In flinto it appears that label and amount is right aligned. I will make then center.

File: src/org/missionassetfund/apps/android/activities/GoalDetailsActivity.java (hunk 9 above)
Comment (amitbharadwaj) [inline src/org/missionassetfund/apps/android/activities/GoalDetailsActivity.java]: yep. Will use that.

**Comment by amitbharadwaj** [comment]
@fcduarte Didn't try jar. They don't provide jar. I got source from github. Didn't try to create jar and then use it.

**Comment by fcduarte** [comment]
The jar is on their maven repository. Let me know if you need assistance to
get it.

**Comment by fcduarte** [comment]
@amitbharadwaj never mind .. I've looked maven and it is not a jar, it's a deprecated apklib file

**Comment by amitbharadwaj** [comment]
@josemontesdeoca @fcduarte I just pushed with category to goal payment. Please review and leave comments. 

Please go ahead and merge the branch if there are not major issues (like crashes). I will address you comments in next PR. This will help test the flow.

**Comment by josemontesdeoca** [comment]
Merging this now.

**Pull request merged**